crates/libcontainer/src/container/init_builder.rs

```diff
@@ -87,6 +87,11 @@ impl InitContainerBuilder {
         };
 
         let user_ns_config = UserNamespaceConfig::new(&spec)?;
+
+        if utils::rootless_required() && !utils::is_in_new_userns() && user_ns_config.is_none() {
+            return Err(LibcontainerError::NoUserNamespace);
+        }
```

Comment:
Can we cover this case in the test?

Reply:
Since I feel  `validate_spec` has a role to check that a spec is valid, how about moving there?

Reply:
Hey, sure :+1: I'll add tests to cover this case, as well as look into moving it into the `validate_spec` function instead . Thanks!

Reply:
Thanks @YJDoc2

Reply:
Done :+1: